File: python/14_LongestCommonPrefix.py
from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        ret = ""
        for i, s in enumerate(strs[0]):
            c = ""
            for j, v in enumerate(strs):
                if i >= len(v):
                    return ret
                if j == 0:
                    c = v[i]
                elif c != v[i]:
                    return ret
            ret += c
        return ret

File: go/14_LongestCommonPrefix.go
package main

func longestCommonPrefix(strs []string) string {
    if len(strs) == 0 {
        return ""
    }
    var ret string
    for i := 0; i <= len(strs[0]); i++ {
        var c string
        for index, v := range strs {
            if i >= len(v) {
                return ret
            }
            if index == 0 {
                c = string(v[i])
            } else if c != string(v[i]) {
                return ret
            }
        }
        ret += c
    }
    return ret
}
